include/nbl/builtin/hlsl/spirv_intrinsics/core.hlsl

```diff
@@ -47,199 +69,1272 @@ static const uint32_t3 LocalInvocationId;
+[[vk::ext_instruction(spv::OpGroupNonUniformBallot)]]
+uint32_t4 groupNonUniformBallot(uint32_t executionScope, bool predicate);
+
+[[vk::ext_capability(spv::CapabilityGroupNonUniformBallot)]]
+[[vk::ext_instruction(spv::OpGroupNonUniformInverseBallot)]]
+bool groupNonUniformInverseBallot(uint32_t executionScope, uint32_t4 value);
+
+[[vk::ext_capability(spv::CapabilityGroupNonUniformBallot)]]
+[[vk::ext_instruction(spv::OpGroupNonUniformBallotBitExtract)]]
+bool groupNonUniformBallotBitExtract(uint32_t executionScope, uint32_t4 value, uint32_t index);
+
+[[vk::ext_capability(spv::CapabilityGroupNonUniformBallot)]]
+[[vk::ext_instruction(spv::OpGroupNonUniformBallotBitCount)]]
+uint32_t groupNonUniformBallotBitCount(uint32_t executionScope, [[vk::ext_literal]] uint32_t operation, uint32_t4 value);
+
+[[vk::ext_capability(spv::CapabilityGroupNonUniformBallot)]]
+[[vk::ext_instruction(spv::OpGroupNonUniformBallotFindLSB)]]
+uint32_t groupNonUniformBallotFindLSB(uint32_t executionScope, uint32_t4 value);
+
+[[vk::ext_capability(spv::CapabilityGroupNonUniformBallot)]]
+[[vk::ext_instruction(spv::OpGroupNonUniformBallotFindMSB)]]
+uint32_t groupNonUniformBallotFindMSB(uint32_t executionScope, uint32_t4 value);
 
 template<typename T>
-[[vk::ext_instruction(spv::OpAtomicCompareExchange)]]
-T atomicCompareExchange([[vk::ext_reference]] T ptr, uint32_t memoryScope, uint32_t memSemanticsEqual, uint32_t memSemanticsUnequal, T value, T comparator);
+[[vk::ext_capability(spv::CapabilityGroupNonUniformShuffle)]]
+[[vk::ext_instruction(spv::OpGroupNonUniformShuffle)]]
+T groupNonUniformShuffle(uint32_t executionScope, T value, uint32_t id);
 
-template<typename T, typename Ptr_T> // DXC Workaround
-[[vk::ext_instruction(spv::OpAtomicCompareExchange)]]
-enable_if_t<is_spirv_type_v<Ptr_T>, T> atomicCompareExchange(Ptr_T ptr, uint32_t memoryScope, uint32_t memSemanticsEqual, uint32_t memSemanticsUnequal, T value, T comparator);
+template<typename T>
+[[vk::ext_capability(spv::CapabilityGroupNonUniformShuffle)]]
+[[vk::ext_instruction(spv::OpGroupNonUniformShuffleXor)]]
+T groupNonUniformShuffleXor(uint32_t executionScope, T value, uint32_t mask);
 
+template<typename T>
+[[vk::ext_capability(spv::CapabilityGroupNonUniformShuffleRelative)]]
+[[vk::ext_instruction(spv::OpGroupNonUniformShuffleUp)]]
+T groupNonUniformShuffleUp(uint32_t executionScope, T value, uint32_t delta);
 
-template<typename T, uint32_t alignment>
-[[vk::ext_capability(spv::CapabilityPhysicalStorageBufferAddresses)]]
-[[vk::ext_instruction(spv::OpLoad)]]
-T load(pointer_t<spv::StorageClassPhysicalStorageBuffer,T> pointer, [[vk::ext_literal]] uint32_t __aligned = /*Aligned*/0x00000002, [[vk::ext_literal]] uint32_t __alignment = alignment);
+template<typename T>
+[[vk::ext_capability(spv::CapabilityGroupNonUniformShuffleRelative)]]
+[[vk::ext_instruction(spv::OpGroupNonUniformShuffleDown)]]
+T groupNonUniformShuffleDown(uint32_t executionScope, T value, uint32_t delta);
```

Comment:
does execution scope need to be a literal?

Reply:
Nop. just checked